docs/source/java/cdata.rst

```diff
@@ -0,0 +1,313 @@
+        const int64_t* data = int64_array->raw_values();
+        std::cout << "Data To Exchange Pointer = " << data << std::endl;
+        for (int j = 0; j < int64_array->length(); j++){
+            std::cout << "Data To Exchange Value[" << j << "] = " << data[j] << std::endl;
+        }
+
+        *array = (struct ArrowArray) {
+                // Data description
+                .length = int64_array->length(),
+                .offset = 0,
+                .null_count = 0,
+                .n_buffers = 2,
+                .n_children = 0,
+                .children = NULL,
+                .dictionary = NULL,
+                // Bookkeeping
+                .release = &release_int32_array
+        };
+
+        // Allocate list of buffers
+        array->buffers = (const void**) malloc(sizeof(void*) * array->n_buffers);
+        assert(array->buffers != NULL);
+        array->buffers[0] = NULL;  // no nulls, null bitmap can be omitted
+        array->buffers[1] = data;
+
+        std::cout << "C Data - Array Pointer = " << array << std::endl;
+        std::cout << "C Data - Array Data Pointer Buffer array->buffers[1] = " << array->buffers[1] << std::endl;
+    }
+
+**Java Side**
+
+Define Java code CDataInterfaceLibraryConfig.java that consume by JNI C++ functions exported through
+C Data Interface:
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <project xmlns="http://maven.apache.org/POM/4.0.0"
+             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+             xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+        <modelVersion>4.0.0</modelVersion>
+        <groupId>org.example</groupId>
+        <artifactId>cpp-java-cdata</artifactId>
+        <version>1.0-SNAPSHOT</version>
+        <properties>
+            <maven.compiler.source>8</maven.compiler.source>
+            <maven.compiler.target>8</maven.compiler.target>
+        </properties>
+        <dependencies>
+            <dependency>
+                <groupId>org.bytedeco</groupId>
+                <artifactId>javacpp</artifactId>
+                <version>1.5.7</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.arrow</groupId>
+                <artifactId>arrow-c-data</artifactId>
+                <version>7.0.0</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.arrow</groupId>
+                <artifactId>arrow-memory-netty</artifactId>
+                <version>7.0.0</version>
+            </dependency>
+        </dependencies>
+    </project>
+
+.. code-block:: java
+
+    import org.bytedeco.javacpp.annotation.Platform;
+    import org.bytedeco.javacpp.annotation.Properties;
+    import org.bytedeco.javacpp.tools.InfoMap;
+    import org.bytedeco.javacpp.tools.InfoMapper;
+
+    @Properties(
+            target = "CDataInterfaceLibrary",
```

Comment:
I would rather name this something like "CDataInterfaceExample"

Reply:
Changed